kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/KubernetesCrudDispatcher.java

```diff
@@ -233,7 +237,15 @@ public MockResponse handleWatch(String path) {
     if (resourceName != null) {
       query = query.add(new Attribute("name", resourceName));
     }
-    WatchEventsListener watchEventListener = new WatchEventsListener(context, query, watchEventListeners, LOGGER);
+    WatchEventsListener watchEventListener = new WatchEventsListener(context, query, watchEventListeners, LOGGER) {
+      @Override
+      public void onOpen(WebSocket webSocket, Response response) {
+        super.onOpen(webSocket, response);
+        map.entrySet().stream()
+          .filter(entry -> this.attributeMatches(entry.getKey()))
+          .forEach(entry -> this.sendWebSocketResponse(entry.getValue(), ADDED));
```

Comment:
I see `WatchEventsListener` is only used here. Is there specific reason you are overriding `onOpen` implementation instead of adding this logic in `WatchEventsListener#onOpen`?

Reply:
I could go either way on that.  It seemed more targeted to override in place.  I can alternatively do this with passing an onOpen consumer in the constructor.